test/e2e/alertmanager_e2e_test.go

```diff
@@ -188,3 +191,71 @@ func TestExposingAlertmanagerWithIngress(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestMeshInitialization(t *testing.T) {
+	var amountAlertmanagers int32 = 3
+	alertmanager := &v1alpha1.Alertmanager{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-alertmanager",
+		},
+		Spec: v1alpha1.AlertmanagerSpec{
+			Replicas: &amountAlertmanagers,
+			Version:  "master",
+		},
+	}
+
+	alertmanagerService := framework.MakeAlertmanagerService(alertmanager.Name, "alertmanager-service", v1.ServiceTypeClusterIP)
+
+	defer func() {
+		if err := framework.DeleteAlertmanagerAndWaitUntilGone(alertmanager.Name); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.DeleteService(alertmanagerService.Name); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	if err := framework.CreateAlertmanagerAndWaitUntilReady(alertmanager); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := framework.CreateServiceAndWaitUntilReady(alertmanagerService); err != nil {
+		t.Fatal(err)
+	}
+
+	proxyGet := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).ProxyGet
+	err := framework.Poll(time.Second*20, time.Second, func() (bool, error) {
+		request := proxyGet("", alertmanagerService.Name, "web", "/api/v1/status", make(map[string]string))
```

Comment:
requesting a random instance load balanced through a service may not be the best thing to do here, I think we should rather test that each alertmanager instance has discovered all other peers, so instead of testing this for one instance we should test it for all